data_subscriptions/api/resources/subscription.py

```diff
@@ -27,6 +27,10 @@ def can_subscribe(user_id, dataset_id=False):
         ).scalar()
 
 
+def get_subscription_by_dataset_id(dataset_id):
+    return db.session.query(Model).filter(Model.dataset_id == dataset_id).first()
```

Comment:
@luccasmmg not sure I know this API but `.first()` method would probably get the first result in the list? I'm just wondering if this method suppose to get all subscriptions for given dataset?

Reply:
You are most definetly right, the way that is setup right now would create a bunch of problems once there were more than two users subscribing to the same dataset